src/main/java/org/opensearch/ad/model/AnomalyDetector.java

```diff
@@ -266,6 +279,7 @@ public void writeTo(StreamOutput output) throws IOException {
         } else {
             output.writeBoolean(false);
         }
+        output.writeOptionalString(resultIndex);
```

Comment:
`AnomalyDetector` used in these transport requests/responses
```
ADBatchAnomalyResultRequest, we only send to nodes with same AD version
ForwardADTaskRequest,  we only send to nodes with same AD version
IndexAnomalyDetectorRequest / IndexAnomalyDetectorResponse,  send to local node 
PreviewAnomalyDetectorRequest / PreviewAnomalyDetectorResponse, send to local node
GetAnomalyDetectorResponse,  request only sent to local node
```

Reply:
Real time detector does not send AnomalyDetector across the wire.  Is the code in the reader part still needed:

```
public AnomalyDetector(StreamInput input) throws IOException {
...
if (input.available() > 0) {
            resultIndex = input.readOptionalString();
        }
```

Reply:
Thanks for confirming. Ideally we don't need this. How about we keep this now in case we missed any edge case, this could help not reading beyond the input stream.

Reply:
Can you add comments on why we need bwc in stream read but not in stream write? Otherwise, it may cause confusion for later maintenance. The same applies to AnomalyDetectorJob.

Reply:
Did more checking, don't see any BWC issue for AnomalyDetector, will remove this checking `if (input.available() > 0) {`.